engine/access/rpc/backend/backend_scripts.go

```diff
@@ -81,10 +84,45 @@ func (b *backendScripts) executeScriptOnExecutionNode(
 		Arguments: arguments,
 	}
 
-	execResp, err := b.executionRPC.ExecuteScriptAtBlockID(ctx, &execReq)
+	// if an executionRPC is provided, send the script to that execution node
+	if b.staticExecutionRPC != nil {
```

Comment:
The new behaviour of requesting an execution node based on whether we've seen its execution receipt depends on now having a state execution RPC specified. But it also seems that other RPC methods depend on it being specified (eg: https://github.com/onflow/flow-go/pull/313/files#r562412571)

Reply:
that is correct - eventually, I will port this logic to all such methods (the logic of using the rpc is specified or choosing 3 random exe nodes)

Reply:
I mean it seems like only or the other will work currently. Either we specify a static execution node and it's used for all RPC methods, or we don't specify a static execution node and we'll use the new `executionNodesForBlockID` for `ExecuteScript` but the other RPC methods will throw a nil pointer error.

Reply:
the `staticExecutionRPC` will soon not be needed for any EN related calls - I am in the process of making it optional for `Transactions` as well (the last place where it is mandatory).

Reply:
sorry if that looked confusing - I was pushing commits as I made them but eventually no one should error if `staticExecutionRPC` is not defined.